sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -181,10 +183,15 @@ class AstBuilder extends DataTypeAstBuilder with SQLConfHelper
         if bl.multipartIdentifier().getText.nonEmpty &&
           bl.multipartIdentifier().getText.toLowerCase(Locale.ROOT) !=
               el.multipartIdentifier().getText.toLowerCase(Locale.ROOT) =>
-        throw SqlScriptingErrors.labelsMismatch(
-          bl.multipartIdentifier().getText, el.multipartIdentifier().getText)
+        withOrigin(bl) {
+          throw SqlScriptingErrors.labelsMismatch(
+            CurrentOrigin.get, bl.multipartIdentifier().getText, el.multipartIdentifier().getText)
```

Comment:
```suggestion
            bl, bl.multipartIdentifier().getText, el.multipartIdentifier().getText)
```

Reply:
Sorry, but I dont understand what you mean by this, your suggestion doesn't compile because bl is BeginLabelContext, not Origin, that's why we included withOrigin there.